ingredient/ingredient-kubernetes/src/plugin.ts

```diff
@@ -0,0 +1,98 @@
+import fs from 'fs'
+import { BaseIngredient, BakeVariable } from "@azbake/core"
+import { promisify } from 'util';
+import { exec as exec_from_child_process } from 'child_process';
+import Replace from 'replace-in-file';
+const exec = promisify(exec_from_child_process);
+
+export class KubernetesPlugin extends BaseIngredient {
+
+    public async Execute(): Promise<void> {
+        const kubeconfigFilename = 'kubeconfig.yaml'
+        try {
+            let k8sYamlPath = await this._ingredient.properties.source.valueAsync(this._ctx);
+            if (!await promisify(fs.exists)(k8sYamlPath)) {
+                throw "file/path not found: " + k8sYamlPath
+            }
+            await this.replaceTokens(k8sYamlPath);
+            let testDeployment = this._ingredient.properties.parameters.get("testDeployment");
+            let kubeConfigParam = await this.getKubeConfigParameter(kubeconfigFilename);
+            try {
+                let { stdout } = await exec(`kubectl apply ${kubeConfigParam} -f ${k8sYamlPath}`);
+                this._logger.log(`stdout: ${stdout}`);
+                if (testDeployment && await testDeployment.valueAsync(this._ctx)) {
+                    ({ stdout } = await exec(`kubectl get ${kubeConfigParam} services`));
+                    this._logger.log(`stdout: ${stdout}`);
+                    ({ stdout } = await exec(`kubectl.exe delete ${kubeConfigParam} -f ${k8sYamlPath}`));
+                    this._logger.log(`stdout: ${stdout}`);
+                }
+            } finally {
+                if (kubeConfigParam) {
+                    try {
+                        await promisify(fs.unlink)(kubeconfigFilename)
+                    } catch (error) {
+                        this._logger.error(error)
+                    }
+                }
+            }
+        } catch (error) {
+            this._logger.error('deployment failed: ' + error)
+            throw error
+        }
+    }
+    private async getKubeConfigParameter(kubeconfigFilename: string) {
+        let configParam = "";
+        let b64KubeConfigContent = this._ctx.Environment.variables.get("b64KubeConfigContent");

```

Comment:
Why are you looking for a hard coded global variable? Just pass in the value as a param to the ingredient, so that a recipe can contain multiple configs if they need to deploy to different clusters/rbac per ingredient.

i.e.  params:
           -config : ["coreutils.variable('my_config_base64_variable')"]